### ISSUE TITLE
Feature Request: Flag to disable automatic background download of updates

### ISSUE THREAD

**Issue by zencephalon** [meteor/meteor#7372]
TL;DR: Add a `--no-background-update` flag to meteor-tool

Currently keeping several small Meteor applications running in parallel on a commodity hosting service such as Linode works great, most of the time. The only time my Meteor applications crash happens because every application will simultaneously try to download new releases of Meteor, causing them to all run out of memory. They can also manage to use up all the space allocated to `/tmp` because they can't clean up after themselves when crashing due to `ENOMEM`.

```
=> Started proxy.

/home/zen/.meteor/packages/coffeescript/.1.0.17.1xqchvt++os+web.browser+web.cordova/plugin.compileCoffeescript.os/npm/node_modules/meteor/promise/node_modules/meteor-promise/promise_server.js:116
      throw error;
            ^
Error: spawn ENOMEM
    at errnoException (child_process.js:1011:11)
    at ChildProcess.spawn (child_process.js:958:11)
    at Object.exports.spawn (child_process.js:746:9)
    at AppProcess._spawn (/tools/runners/run-app.js:264:31)
    at AppProcess.start (/tools/runners/run-app.js:86:22)
    at AppRunner._runOnce (/tools/runners/run-app.js:761:16)
    at AppRunner._fiber (/tools/runners/run-app.js:887:28)
    at /tools/runners/run-app.js:411:12
```

This has caused me to dread Meteor updates as they also require me to go into the server and get things running again. Quite unfortunate given the amazing innovations coming out in the latest releases.

I understand that I could avoid this problem by compiling a Node bundle, however installing a compatible version of Node.js has proven prohibitively difficult on certain platforms. Further, I enjoy the benefits of `git pull` as a deployment strategy in combination with Meteor's reloading.

I truly love that I can easily run Meteor on just about any server and using Nginx get a lot of bang for my buck. I highly recommend Meteor to my friends for that reason and all the other wonderful quality of life improvements it has brought.

I understand that Galaxy can also solve these problems but I bet a lot of other people also appreciate that Meteor works on their cheapo servers just fine.

I would love to run `meteor --production --no-background-update` and rest in peace knowing my apps will stay online through Meteor version releases and that I can update on my own schedule.

**Comment by laosb** [comment]
`meteor --production` is not for production. It's just for *simulating* production. This may affect your app's security, so please don't use `--production` for it.

Anyway, on some machines downloading in background does leads to `spawn ENOMEM`. We should have a way to disable it. See discussions there: #7026.

Closing.

PS: I would suggest adding a warning when use `--production` to persuade users not to use this on production.
